projects/ngx-sub-form/src/lib/ngx-sub-form.types.ts

```diff
@@ -40,3 +40,9 @@ export interface FormGroupOptions<T> {
    */
   updateOn?: 'change' | 'blur' | 'submit';
 }
+
+// Unfortunately due to https://github.com/microsoft/TypeScript/issues/13995#issuecomment-504664533 the initial value
+// cannot be fully type narrowed to the exact type that will be passed.
+export interface NgxFormWithArrayControls<T> {
+  createFormArrayControl(key: ArrayPropertyOf<T>, initialValue: ArrayTypeOfPropertyOf<T>): FormControl;
```

Comment:
I think that `initialValue: ArrayTypeOfPropertyOf<T>` should rather be `initialValue: ArrayTypeOfPropertyOf<T> | null`

Reply:
So the reason I switched this to be an interface was to remove the restriction that the array sub type had to be assignable to null - with my PR there is no way that the value can be null unless that is what the parent component is passing in, in which case it would have to be on the user type.

Reply:
Make sense :)